### PR TITLE
Add listRoles and dropRole to the client.go

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:
Adds new endpoints listRole and dropRole to the client.go once they're added to the mgmt-api

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/httphelper/client.go [hunk 1]
```diff
@@ -324,6 +324,78 @@ func (client *NodeMgmtClient) CallCreateRoleEndpoint(pod *corev1.Pod, username s
 	return nil
 }
 
+// Create a new superuser with the given username and password
```
Comment (adejanovski) [inline pkg/httphelper/client.go]: **Issue:** Comment needs an update to match the function

File: pkg/httphelper/client.go [hunk 2]
```diff
@@ -324,6 +324,78 @@ func (client *NodeMgmtClient) CallCreateRoleEndpoint(pod *corev1.Pod, username s
...
+func parseListRoles(body []byte) ([]User, error) {
+	var users []User
+	if err := json.Unmarshal(body, &users); err != nil {
+		fmt.Printf("Received an error: %v\n", err)
+		return nil, err
+	}
+	return users, nil
+}
+
+// Create a new superuser with the given username and password
```
Comment (adejanovski) [inline pkg/httphelper/client.go]: **Issue:** Comment needs an update to match the function

File: pkg/httphelper/client.go [hunk 3]
```diff
@@ -324,6 +324,78 @@ func (client *NodeMgmtClient) CallCreateRoleEndpoint(pod *corev1.Pod, username s
 	return nil
 }
 
+// Create a new superuser with the given username and password
+func (client *NodeMgmtClient) CallDropRoleEndpoint(pod *corev1.Pod, username string) error {
```
Comment (adejanovski) [inline pkg/httphelper/client.go:332]: **Todo:** Could you add a test for this deletion endpoint? We have one for the list endpoint only it seems.

**Review by adejanovski — CHANGES_REQUESTED** [review]

File: pkg/httphelper/client.go (hunk 3 above)
Comment (burmanm) [reply pkg/httphelper/client.go:332]: Done (for all 3 methods, even the one I didn't add in this PR).

**Review by adejanovski — APPROVED** [review]

**Pull request merged**